### PR TITLE
Adding cashier closing feature 

### PR DESCRIPTION
**Pull request by elba7r**
As it is very important for the cashiers to have a cashier closing feature i have added the doctype at the accounts module and put a link for it at the pos menu also a link for it at the accounts py config file to show it at the accounts module page .

The cashier closing takes the date time and the user automatically, the user can add the expenses which is taken from his save and add the custody which he received at the beginning of his shift also he add the amount which is at his save at the end of the shift and the system will calculate the net total amount which is supposed to be equal to the total of his sales invoices.

### CODE REVIEW

File: erpnext/accounts/doctype/cashier_closing/cashier_closing.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+// License: GNU General Public License v3. See license.txt
+
+frappe.ui.form.on('Cashier Closing', {
+
+	expense: function(frm){
```
Comment (rmehta) [inline erpnext/accounts/doctype/cashier_closing/cashier_closing.js]: all calculations must be on server side

https://github.com/frappe/erpnext/wiki/Coding-Standards#all-business-logic-must-be-api-friendly-and-must-be-implemented-in-server-side-python

**Review by rmehta — CHANGES_REQUESTED** [review]
Please also add screenshots for UX

https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#user-experience--animated-gif

**Comment by nabinhait** [comment]
@elba7r How this is different from "POS Closing Voucher" feature in develop branch?
https://github.com/frappe/erpnext/pull/14270

File: erpnext/accounts/doctype/cashier_closing/cashier_closing.js (hunk 1 above)
Comment (elba7r) [reply erpnext/accounts/doctype/cashier_closing/cashier_closing.js]: now the calculations are on the server side

**Comment by elba7r** [comment]
@rmehta  @nabinhait 

- Now all calculations are at the server side
- The Current "POS Closing Voucher"  is working as a summary for the cashier invoices and i believe it is not the aim from the cashier closing as i can get that information from the reporting of the sales invoices, but the cashier closing needed for the cashier him self to be able to write the values by him self then the manager come and review and compare it with the invoices recorded in the data base to be able to find the difference  if found .
-cashier closing feature get the date automatic and the user also, the cashier set the From Date and To Date by him self ( in case he worked more than one shift )
-he also write the expenses which token from his save and the custody which he got at the beginning of his shift ( small changes to help him giving change to the customer or any else) 
-he also add the amount of money based on mode of payments ( cash - credit card ... etc)
-the system calculate the outstanding amounts on the period ( from time - to time ) for the user and add it automatically 
The system making the calculations ( net_amount = payments + expenses + outstanding - custody ) and it should equal the total of his sales invoices, 
so if the manager found any differences he can track and check for the reasons

cashier closing 1
cashier closing 2

**Comment by yashodhank** [comment]
@zkhan44499 This is not right place to seek help, go to https://discuss.erpnext.com and ask for help or hire service provider https://erpnext.org/service-providers.

**Comment by f-9t9it** [comment]
Hello,

We have made Enhanced Pos Closing Voucher, Using a method for Cash Opening + Sales - Returns - Cash Change given = Closing Cash and also given a Manual Fetch button to fetch updated sales invoice incase the user opened pos closing voucher in the middle of the day.

I can share our Pos Closing Voucher screenshots and also we designed receipt printer size print format. 
Maybe this can help.
In our next pos closing voucher update will have expense added to it.

**Comment by nabinhait** [comment]
@f-9t9it Cool 👍 Let's send the PR with some screenshots.

**Comment by f-9t9it** [comment]
whatsapp image 2019-01-31 at 1 05 52 pm
pos closing -1
pos closing -2

Kindly find some for now. Need to edit 2 more.

**Comment by f-9t9it** [comment]
pos closing - fetch updates

pos closing - print format

**Pull request merged**